apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.php

```diff
@@ -40,45 +48,94 @@ function add_crossorigin_to_script_els( $tag ) {
 	return $tag;
 }
 
+/**
+ * Temporary function to feature flag Sentry by segment. We'll be testing
+ * it on production (simple sites) for a while to see if it's feasible to
+ * activate it for all sites and perhaps get rid of our custom solution.
+ * If it works well, we'll activate for all simple sites and look into.
+ * activating it for WoA, too.
+ *
+ * @param int $user_id the user id.
+ * @return bool
+ */
+function user_in_sentry_test_segment( $user_id ) {
+	$current_segment = 1; // segment of existing users that will get this feature in %.
+	$user_segment    = $user_id % 100;
+
+	// We get the last two digits of the user id and that will be used to decide in what
+	// segment the user is. i.e if current_segment is 10, then only ids that end in < 10
+	// will be considered part of the segment.
+	return $user_segment < $current_segment;
+}
+
+/**
+ * Returns whether or not the site loading ETK is in the WoA env.
+ *
+ * @return bool
+ */
+function is_atomic() {
+	return defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+}
+/**
+ * Return whether Sentry should be activated for a given user.
+ *
+ * In this phase, a12s have the possibility of configuring what error reporter to choose
+ * through the sticker. a12s shouldn not be covered by the segment logic.
+ *
+ * Regular users have the error reporter chosen based on the segmentation logic, only.
+ *
+ * @param int $user_id The user id. Used to check if the user is A8C or in
+ * the Sentry test segment.
+ * @param int $blog_id The blog ID. Usually the value of `get_current_blog_id`.
+ * Used to check if the sticker is applied if user is A8C.
+ */
+function should_activate_sentry( $user_id, $blog_id ) {
+	return ( is_automattician( $user_id ) && has_blog_sticker( 'error-reporting-use-sentry', $blog_id ) )
+		|| ( ! is_automattician( $user_id ) && user_in_sentry_test_segment( $user_id ) );
+}
+
 /**
  * Enqueue assets
  */
 function enqueue_script() {
 	$asset_file          = include plugin_dir_path( __FILE__ ) . 'dist/error-reporting.asset.php';
 	$script_dependencies = isset( $asset_file['dependencies'] ) ? $asset_file['dependencies'] : array();
 	$script_version      = isset( $asset_file['version'] ) ? $asset_file['version'] : filemtime( plugin_dir_path( __FILE__ ) . 'dist/error-reporting.js' );
+	$script_id           = 'a8c-fse-error-reporting-script';
 
 	wp_enqueue_script(
-		'a8c-fse-error-reporting-script',
+		$script_id,
 		plugins_url( 'dist/error-reporting.js', __FILE__ ),
 		$script_dependencies,
 		$script_version,
 		true
 	);
+
+	wp_localize_script(
+		$script_id,
+		'dataFromPHP',
```

Comment:
I think we should make this more unique, since we're setting a variable in the global window -- theoretically, any plugin in PHP could decide to use this name for their data and it might conflict.

Reply:
Ah, good point. I'll add a prefix.